Dockerfile

```diff
@@ -1,18 +1,46 @@
-# syntax=docker.io/docker/dockerfile:1.4
+# syntax=docker.io/docker/dockerfile:1
+
+# This enforces that the packages downloaded from the repositories are the same
+# for the defined date, no matter when the image is built.
+ARG NOBLE_DATE=20240801
+ARG APT_UPDATE_SNAPSHOT=${NOBLE_DATE}T030400Z
 
 ################################################################################
-# cross build stage
-FROM ubuntu:noble-20240801 as build-stage
+# cross base stage
+FROM ubuntu:noble-${NOBLE_DATE} AS base-build-stage
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN <<EOF
+set -e
+apt update
+apt install -y --no-install-recommends ca-certificates
+apt update --snapshot=${APT_UPDATE_SNAPSHOT}
+EOF
+
+################################################################################
+# riscv64 base stage
+FROM --platform=linux/riscv64 ubuntu:noble-${NOBLE_DATE} as base-target-stage
```

Comment:
Docker is raising a warning that the `as` keyword should match the case of the `from` keyword.

Reply:
sent fixup